private-templates-service/server/src/db/providers/InMemoryDBProvider.ts

```diff
@@ -0,0 +1,249 @@
+// const this.usersPath: string = path.join(dbFilePath, 'users.json');
+// const this.templatesPath: string = path.join(dbFilePath, 'templates.json');
+const options = {
+    encoding: 'utf8'
+}; 
+const SUCCES_WRITE_MESSAGE: string = "Succesfully written to %s DB";
+const SUCCES_IMPORT_MESSAGE: string = "Succesfully imported from %s DB";
+const USER_COLLECTION_NAME: string = "User";
+const TEMPLATE_COLLECTION_NAME: string = "Template";
+
+
+// uuidv4
+function generateUniqueID() : string{
+    return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function(c) {
+        var r = Math.random() * 16 | 0, v = c == 'x' ? r : (r & 0x3 | 0x8);
+        return v.toString(16);
+      });
+  }
+
+export class InMemoryDBProvider extends StorageProvider {
+
+    users: Map<string, Collections.User> = new Map();
+    templates: Map<string, Collections.Template> = new Map();
+    readTryCount: number = 0;
+    writeTryCount: number = 0;
+    usersPath: string;
+    templatesPath: string;
+    
+    constructor(dbDirectoryPath: string) {
+        super("");
+        this.usersPath = path.join(dbDirectoryPath, 'users.json');
+        this.templatesPath = path.join(dbDirectoryPath, 'templates.json');
+        
+    }
+
+
+    private async _insert<T extends Collections.Template | Collections.User>(doc: T,
+                                                                             collection: Map<String, T>,
+                                                                             commitFunction?:() => Promise<void>,
+                                                                             ifCommit: boolean = true, )
+                                                                             : Promise<void> {
+        this._checkID(doc);
+        if (!collection.has(doc._id!)) {
+            collection.set(doc._id!, doc);
+            if(ifCommit && commitFunction) {
+                commitFunction();
+            }
+        } else {
+            logger.error("Object with id: " + doc._id + " already exists. Insertion failed");
+        }
+    }
+
+    private async _writeData(path: string, data: string, successMessage: string, options: Object) : Promise<void> {
+        return fs.writeFile(path, data, options)
+        .then(result => console.log(successMessage))
+        .catch(err => {
+            if(this.writeTryCount < MAX_TRY_COUNT) {
+                logger.error(sprintf("Cannot write to %s. Erros is %s.", path, err));
+                // TODO: set timeout to retry write
+                // setTimeout(this._writeData, 1000); 
+            }
+        });
+    }
+
+    private async _readData(path:string, successMessage: string, options: Object) : Promise<string | void> {
+        return fs.readFile(path, options)
+        .then(buffer => {return buffer.toString()})
+        .catch(err => {
+            logger.error(sprintf("Cannot read from %s. Erros is %s.", path, err));
```

Comment:
I think you can format directly through the logger: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/bunyan/index.d.ts#L170